pkg/auth/providers/oidc/oidc_client.go

```diff
@@ -22,19 +22,21 @@ func (o *OpenIDCProvider) AddCertKeyToContext(ctx *context.Context, certificate,
 }
 
 func GetClientWithCertKey(httpClient *http.Client, certificate, key string) error {
-	keyPair, err := tls.X509KeyPair([]byte(certificate), []byte(key))
-	if err != nil {
-		return err
-	}
-	caCertPool := x509.NewCertPool()
-	caCertPool.AppendCertsFromPEM([]byte(certificate))
-	httpClient = &http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{
-				RootCAs:      caCertPool,
-				Certificates: []tls.Certificate{keyPair},
+	if certificate != "" && key != "" {
+		keyPair, err := tls.X509KeyPair([]byte(certificate), []byte(key))
+		if err != nil {
+			return err
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM([]byte(certificate))
+		httpClient = &http.Client{
+			Transport: &http.Transport{
+				TLSClientConfig: &tls.Config{
+					RootCAs:      caCertPool,
+					Certificates: []tls.Certificate{keyPair},
+				},
```

Comment:
We are replacing the httpClient reference by a new one everytime GetClientWithCertKey is called, why do we do this? Also this means you are replacing the k.httpClient reference, this can cause issues if two parallel requests using k.httpClient are made?

Reply:
Agree, seems like a one-time setup should happen every time certs are added or changed. Why'd we move away from that ?

Reply:
I believe I resolved this by only setting the Transport config on the client. Ready for re-review

Reply:
But it will be set again per request and  httpClient is a common instance. Either we should just create a new httpClient per request or Save the cert&key on the httpClient in the TestAndApply method where you saveOIDCConfig during auth setup.

Reply:
I am not building a httpclient per request and adding the auth just to the transport config as to not override any additional settings.